gst/nnstreamer/tensor_query/tensor_query_common.c

```diff
@@ -60,32 +57,422 @@ typedef struct
   {
     struct
     {
-      GSocket *socket;
+      GSocket *client_socket;
       GCancellable *cancellable;
     };
   };
 } TensorQueryClientData;
 
+/**
+ * @brief Connection info structure
+ */
+typedef struct
+{
+  TensorQueryProtocol protocol;
+  gchar *host;
+  guint32 port;
+  /* network info */
+  union
```

Comment:
Is this required to support other protocols rather than TCP? If it is, IMHO, you could pack those in ```union``` when it is really required.